hcloud/client_handler_retry.go

```diff
@@ -30,13 +34,53 @@ func (h *retryHandler) Do(req *http.Request, v any) (resp *Response, err error)
 			// - request preparation
 			// - network connectivity
 			// - http status code (see [errorHandler])
-			if IsError(err, ErrorCodeConflict) {
-				time.Sleep(h.backoffFunc(retries))
-				retries++
-				continue
+			if ctx.Err() != nil {
+				return resp, ctx.Err()
+			}
```

Comment:
`ctx.Err()` and the returned `err` should be the same indeed, this is only an early return in case the context was canceled/timed out. This might be confusing indeed, I'll update the return statements, but I think the early return is still useful.